jdaviz/configs/imviz/plugins/catalogs/catalogs.py

```diff
@@ -107,6 +137,18 @@ def _file_parser(path):
 
         return '', {path: table}
 
+    def _on_catalog_select_click_event(self, msg):
+        xs, ys = self.table._qtable['x_coord'], self.table._qtable['y_coord']
+        # nearest point
+        distsq = (xs - msg.x)**2 + (ys - msg.y)**2
+        ind = np.argmin(distsq)
+        item = self.table.items[ind]
+        if item in self.table.selected_rows:
+            self.table.selected_rows = [sr for sr in self.table.selected_rows if sr != item]
```

Comment:
We might need to revisit this kind of approach when we have gigantic catalogs, though this is ok for now.

Is there anything preventing us from using a boolean array of the same length as the number of table rows to indicate selected rows, or adding a new(/hidden?) boolean column to the astropy table? Applying numpy masking operations would be easy with that implementation, though if users tried to add rows to the table, we'd need to add logic to, e.g., fill in the hidden columns.

Reply:
Agreed - let's make a tech debt ticket to refactor this a bit as we look towards more general catalogs support.   Do you want that on our end or yours?

Reply:
If eventually you want to actually search by sky coord, there is https://docs.astropy.org/en/stable/coordinates/matchsep.html

Can you comment a bit in the code what is the *intention* of the algorithm, especially the if-else? 

Selecting only first match seems not robust enough if there are overlapping points. Not sure if you want to address that here or open follow-up tickets. User might want to select all the points within search radius of the click instead of just one.